onebone/preprocessing/pd.py

```diff
@@ -0,0 +1,41 @@
+"""Transform PRPS format pd data to PRPD format.
+
+- Author: Hyunjae Kim
+- Contact: hyunjae.kim@onepredict.com
+"""
+
+from typing import Tuple
+
+import numpy as np
+
+
+def ps2pd(ps, range_amp: Tuple[int, int] = (0, 256), resol_amp: int = 128) -> np.ndarray:
+    """
+    Transform prps(phase resolved pulse sequance) to a prpd(phaes resolved partial discharge) by marginalizing time dimension.
+
+    Parameters
+    ----------
+    ps : array_like of shape (n_resolution_phase, n_timestep)
+        The data. Ex: kepco standard=(128, 3600)
+    range_amp : tuple (min, max), default=(0, 256)
+        Measurement range of PD DAQ. Refers to DAQ manufacture.
+    resol_amp : int, default 128
+        Desired resolution of amplitude resolution for transformd prpd.
+
+    Returns
+    -------
+    pd : numpy.ndarray of shape (n_resolution_phase, n_resolution_amplitude)
+        The transformed prpd.
+
+    Examples
+    --------
+    To be filled.
+    """
+    imat = np.eye(resol_amp)
+    imat[0, 0] = 0
+    bins = np.linspace(range_amp[0], range_amp[1], resol_amp)
+    p_digit = np.digitize(ps, bins) - 1
+    p_digit_logit = imat[p_digit]
+    pd = np.sum(p_digit_logit, axis=0)
```

Comment:
`ps`가 1-d array이면 `np.sum(..., axis=0)`을 할 필요가 없을 것 같습니다.
```suggestion
    if len(p_digit_logit.shape) == 1:
        pd = p_digit_logit
    else:
        pd = np.sum(p_digit_logit, axis=0)
```
`ps`를 2차원으로 의도하신 것 같은데 이렇게도 쓸 수 있을것 같습니다.
```suggestion
    if len(p_digit_logit.shape) == 1:
        raise ValueError('`ps` has to be 2-dimensions.')
```

Reply:
PRPS의 데이터 자체가 첫번째 차원이 phase, 두번째 차원이 time 이기 때문에 무조건 2차원입니다. 컬러 이미지가 3차원으로 정의되는것처럼 해당 도메인에서의 상식으로 생각해주시면 됩니다.

Reply:
강휘님이 아래에 제안주신대로 입력이 2차원이 아닐 경우 error를 내는 제약조건이 있으면 더 좋을것 같습니다.

Reply:
> PRPS의 데이터 자체가 첫번째 차원이 phase, 두번째 차원이 time 이기 때문에 무조건 2차원입니다. 컬러 이미지가 3차원으로 정의되는것처럼 해당 도메인에서의 상식으로 생각해주시면 됩니다.

그렇군요, 그러면 `ps`에 대해 2차원 array만 입력될 수 있도록 제약조건을 두는 것이 좋을 것 같습니다.

Reply:
두분 감사합니다. 의견 반영하여 수정하였습니다.